src/OpenTelemetry.AutoInstrumentation/Configurations/FileBasedConfiguration/PropagatorConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Vendors.YamlDotNet.Serialization;
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration;
+
+internal class PropagatorConfiguration
+{
+    /// <summary>
+    /// Gets or sets the composite propagator configuration.
+    /// </summary>
+    [YamlMember(Alias = "composite")]
+    public Dictionary<string, object>? Composite { get; set; }
+
+    /// <summary>
+    /// Gets or sets the composite list for the propagator.
+    /// </summary>
+    [YamlMember(Alias = "composite_list")]
+    public string? CompositeList { get; set; }
+
+    public IReadOnlyList<Propagator> GetEnabledPropagators()
```

Comment:
1. `Distinct()` already does the HashSet logic for you, no need to be explicit. 

2. return early >

```
if (Composite is null && string.IsNullOrEmpty(CompositeList))
{
    return Array.Empty<Propagator>();
}
```

3. ToLowerInvariant() adds extra string allocation. Maybe you can reuse `StringComparison.InvariantCultureIgnoreCase` based logic.

4. Try to verify the list and not to copy it's contents over to another.